### PR TITLE
Fixes for madlib-338/341

### PR DESCRIPTION
**Pull request by aojwang**
1. madlib-341: Check duplicated id values in training/classification table. (utility.sql_in)
2. MADLIB-338: Add check for NULL value. The error message will contain the table name.
3. Add an additional star (/**) to the classify, score, display, clean API.
4. Refine documentation for public API.

### CODE REVIEW

File: methods/decision_tree/src/pg_gp/decision_tree.sql_in [hunk 1]
```diff
@@ -2333,37 +2333,35 @@ $$ LANGUAGE PLPGSQL;
...
- *                                      no value. Result such as [2:3]={2,3}, should be read: 
- *                                      jump['feature value'+1], so in this case there were no 0-value points for 
- *                                      this feature. For value 1 jump to 2; for value 2 jump to 3;
- *  - <tt>is_feature_cont BOOLEAN</tt>  It specifies whether the selected feature is a continuous feature.
- *  - <tt>split_value FLOAT</tt>        For continuous feature, it specifies the split value. Otherwise, 
- *     it is fixed to 0.
+ *  - <tt>cat_size INT</tt>           - Number of data point at this node.
+ *  - <tt>parent_id INT</tt>          - Id of the parent branch.
+ *  - <tt>lmc_nid</tt>                - Leftmost child (lmc) node id of the node represented by the current row.
+ *  - <tt>lmc_fval</tt>               - The feature value which leads to the leftmost child (lmc) node.
```
Comment (cwelton) [inline methods/decision_tree/src/pg_gp/decision_tree.sql_in]: lmc_nid/lmc_fval are a replacement for jump, but that code change is not part of this PR, so this comment change is incorrect.

File: methods/decision_tree/src/pg_gp/decision_tree.sql_in [hunk 2]
```diff
@@ -2617,23 +2625,54 @@ END
...
+ *  - <tt>maxclass INTEGER</tt>       - If forced to make a call for a dominant class 
+ *                                      at a given point this is the selected class.
+ *  - <tt>split_gain FLOAT</tt>       - Information gain computed using entropy (at this 
+ *                                      node), also used to determine termination of the branch.
+ *  - <tt>live INT</tt>               - Indication that the branch is still growing. 1 means "live". 
+ *                                      Exit value may be 1 if number of branches reached before termination condition is met.
+ *  - <tt>cat_size INT</tt>           - Number of data point at this node.
+ *  - <tt>parent_id INT</tt>          - Id of the parent branch.
+ *  - <tt>lmc_nid</tt>                - Leftmost child (lmc) node id of the node represented by the current row.
+ *  - <tt>lmc_fval</tt>               - The feature value which leads to the leftmost child (lmc) node.
```
Comment (cwelton) [inline methods/decision_tree/src/pg_gp/decision_tree.sql_in]: As above, the lmc_nid/lmc_fval change is not part of this PR, so this change to the comments is inaccurate.

File: methods/decision_tree/src/pg_gp/decision_tree.sql_in [hunk 3]
```diff
@@ -2668,18 +2707,56 @@ END
...
```
Comment (cwelton) [inline methods/decision_tree/src/pg_gp/decision_tree.sql_in]: As above, the lmc_nid/lmc_fval change is not part of this PR, so this change to the comments is inaccurate.

File: methods/decision_tree/src/pg_gp/decision_tree.sql_in (hunk 2 above)
Comment (aojwang) [inline methods/decision_tree/src/pg_gp/decision_tree.sql_in]: Thanks. Done.

**Comment by cwelton** [comment]
Portions of this PR were merged into PR-117 which has been merged.

**Comment by cwelton** [comment]
Closing per discussion based on the understanding that this is included in later PR.
